docs/development.md

```diff
@@ -1,26 +1,254 @@
+        nodeSelector:
+            grid/type: Operator
+        tolerations:
+        - effect: NoSchedule
+            key: grid/type
+            operator: Equal
+            value: Operator
+        volumes:
+            - name: agent-config-volume
+            configMap:
+                name: agent-configmap
+    backoffLimit: 0
     ```
-    docker run --rm -v ~/.aws/config:/root/.aws/config -e LAYER_NAME="mock-computation-cpp" build_lambda:latest
 
-    #example:
-    docker run --rm -v ~/.aws/config:/root/.aws/config 7379d2acd853
+    For the examples provided with HTC-Grid, these .yaml files are generated automatically when `make happy-path` command is executed. Each .yaml file is a template (e.g., examples/submissions/k8s_jobs/single-task-test.yaml.tpl) that is being filled in with relevant fields that match the AWS account and deployment configuration. The substitution is done by examples/submissions/k8s_jobs/Makefile and relevant attributes are passed in by the ./Makefile.
+
+    Note, once docker image is built and pushed into registry, the yaml file contains the ``command`` parameter that tells how to launch the client application once container is running.
+
+    The deployment yaml file can be generated by extending existing build process, written manually, or generated in any other way that suited the workload.
+
+3. To launch the client, simply execute the following command.
+
+    ```Bash
+    kubectl apply -f ./generated/<custom.yaml.file>.yaml
     ```
 
-* If required update EKS deployment as new images need to be deployed.
\ No newline at end of file
+
+
+### Running a Client Application locally
+
+<TODO>
+
+## 2. Developing a Worker Lambda Function
+
+
+HTC-Grid uses Amazon Elastic Kubernetes Service (Amazon EKS) as a computational backend. Each engine is a pod running two containers: (i) an Agent and a (ii) Worker Lambda Function.
+
+- The **Agent** provides a connectivity layer between the HTC-Grid and the Worker Lambda container. Agent's responsibilities include: (i) pulling for new tasks, (ii) interacting with the Data Plane (I/O), (iii) sending heartbeats back to Control Plane, and (iv) indicating completion of a task. Note, Agent does not need to be changed when developing new applications on HTC-Grid.
+- The **Lambda container** executes the custom code that performs the computational task. The execution is done locally within the container. The code of lambda function needs to be modified during the development.
+    - Note: depending on the workload it is possible for the Worker Lambda function to access HTC-Grid's Data Plane directly or to access any other external systems as might be required. This functionality is not provided as part of the HTC-Grid.
+
+At the high level the development process involves 4 steps:
+
+1. Write a custom Lambda code for the target workload.
+2. Package all the dependencies into a docker container which also includes custom Lambda runtime
+3. Upload the newly built image.
```

Comment:
To ECR ?

Reply:
Agree, needs clarification, it is actually S3